src/gui/optionsdialog.cpp

```diff
@@ -214,444 +225,221 @@ OptionsDialog::OptionsDialog(IGUIApplication *app, QWidget *parent)
         m_ui->tabSelection->item(i)->setSizeHint(size);
     }
 
-    m_ui->IpFilterRefreshBtn->setIcon(UIThemeManager::instance()->getIcon(u"view-refresh"_qs));
+    connect(m_ui->tabSelection, &QListWidget::currentItemChanged, this, &ThisType::changePage);
 
-    m_ui->labelGlobalRate->setPixmap(Utils::Gui::scaledPixmapSvg(UIThemeManager::instance()->getIconPath(u"slow_off"_qs), this, Utils::Gui::mediumIconSize(this).height()));
-    m_ui->labelAltRate->setPixmap(Utils::Gui::scaledPixmapSvg(UIThemeManager::instance()->getIconPath(u"slow"_qs), this, Utils::Gui::mediumIconSize(this).height()));
+    // Load options
+    loadBehaviorTabOptions();
+    loadDownloadsTabOptions();
+    loadConnectionTabOptions();
+    loadSpeedTabOptions();
+    loadBittorrentTabOptions();
+    loadRSSTabOptions();
+#ifndef DISABLE_WEBUI
+    loadWebUITabOptions();
+#endif
 
-    m_ui->deleteTorrentWarningIcon->setPixmap(QApplication::style()->standardIcon(QStyle::SP_MessageBoxCritical).pixmap(16, 16));
-    m_ui->deleteTorrentWarningIcon->hide();
-    m_ui->deleteTorrentWarningLabel->hide();
-    m_ui->deleteTorrentWarningLabel->setToolTip(u"<html><body><p>" +
-        tr("By enabling these options, you can <strong>irrevocably lose</strong> your .torrent files!") +
-        u"</p><p>" +
-        tr("When these options are enabled, qBittorrent will <strong>delete</strong> .torrent files "
-        "after they were successfully (the first option) or not (the second option) added to its "
-        "download queue. This will be applied <strong>not only</strong> to the files opened via "
-        "&ldquo;Add torrent&rdquo; menu action but to those opened via <strong>file type association</strong> as well") +
-        u"</p><p>" +
-        tr("If you enable the second option (&ldquo;Also when addition is cancelled&rdquo;) the "
-        ".torrent file <strong>will be deleted</strong> even if you press &ldquo;<strong>Cancel</strong>&rdquo; in "
-        "the &ldquo;Add torrent&rdquo; dialog") +
-        u"</p></body></html>");
+    // Load Advanced settings
+    m_advancedSettings = new AdvancedSettings(app, m_ui->tabAdvancedPage);
+    m_ui->advPageLayout->addWidget(m_advancedSettings);
+    connect(m_advancedSettings, &AdvancedSettings::settingsChanged, this, &ThisType::enableApplyButton);
 
-    m_ui->hsplitter->setCollapsible(0, false);
-    m_ui->hsplitter->setCollapsible(1, false);
-    // Get apply button in button box
-    m_applyButton = m_ui->buttonBox->button(QDialogButtonBox::Apply);
+    // setup apply button
+    m_applyButton->setEnabled(false);
```

Comment:
Does it even make sense to have separate member for "Apply" button? Why don't access it via button box when if is needed?

Reply:
Then there will be a function call every time: `m_ui->buttonBox->button(QDialogButtonBox::Apply);` and this is quite long..

Reply:
> Then there will be a function call every time: `m_ui->buttonBox->button(QDialogButtonBox::Apply);` and this code segment is quite long..

Not every time but once per function where it is used (to assign local variable if it is used multiple time in such function), isn't it?

Reply:
>Not every time but once per function where it is used (to assign local variable if it is used multiple time in such function), isn't it?

That is not the common usage here: `OptionsDialog::enableApplyButton()` slot is commonly connected to.

Another viewpoint is, I would prefer have another variable `bool m_dirty` to record whether options has changed. Then it is possible to ditch `m_applyButton` class variable. Anyway this belongs to another PR.